src/main/java/com/google/sps/servlets/DestinationDataServlet.java

```diff
@@ -0,0 +1,152 @@
+            .withHint(request.getParameter(HINT2_PARAMETER))
+            .withHint(request.getParameter(HINT3_PARAMETER))
+            .build();
+
+    /* Retrieves the obscurity level chosen by the user as a List of Strings and converts the level to an Enum Obscurity value */
+    String obscureLevel = request.getParameter(OBSCURITY_PARAMETER);
+    Destination.Obscurity level = convertLevelToEnum(obscureLevel);
+
+    /* Retrieves the tags selected by the user as a List of Strings and converts them into a Set of Enum Tags */
+    List<String> tags =
+        Arrays.stream(request.getParameterValues(TAG_PARAMETER))
+            .filter(tag -> tag != null)
+            .collect(Collectors.toList());
+    Set<Destination.Tag> checkedTags = convertTagsToEnum(tags);
+
+    Destination destination =
+        new Destination.Builder()
+            .withName(name)
+            .withLocation(location)
+            .withCity(city)
+            .withDescription(description)
+            .withRiddle(riddle)
+            .withTags(checkedTags)
+            .withObscurity(level)
+            .build();
+
+    /* TODO: Store the data in datastore rather than locally */
+    destinations.add(destination);
+    /* TODO: redirect back index.html once the data is stored in datastore */
+    response.sendRedirect(REDIRECT_URL);
+  }
+
+  /* Retrieves the most recent destination created by the user, turns it into a JSON formatted string,
+   *and displays the JSON-ified String on /destination-data
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Gson gson = new Gson();
+    String json = gson.toJson(destinations.get(destinations.size() - 1));
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  public Set<Destination.Tag> convertTagsToEnum(List<String> tags) {
+    Set<Destination.Tag> tagEnums = new HashSet<Destination.Tag>();
+    for (String tag : tags) {
+      switch (tag) {
+        case "art":
+          tagEnums.add(Destination.Tag.ART);
+          break;
+        case "sports":
+          tagEnums.add(Destination.Tag.SPORT);
+          break;
+        case "historical":
+          tagEnums.add(Destination.Tag.HISTORICAL);
+          break;
+        case "food":
+          tagEnums.add(Destination.Tag.FOOD);
+          break;
+        case "family":
+          tagEnums.add(Destination.Tag.FAMILY);
+          break;
+        case "tourist":
+          tagEnums.add(Destination.Tag.TOURIST);
+          break;
+      }
+    }
+
+    return tagEnums;
+  }
+
+  public Destination.Obscurity convertLevelToEnum(String obscureLevel) {
+    switch (obscureLevel) {
+      case "easy":
+        return Destination.Obscurity.EASY;
+      case "medium":
+        return Destination.Obscurity.MEDIUM;
+      case "hard":
+        return Destination.Obscurity.HARD;
+    }
+
+    return Destination.Obscurity.UNDEFINED;
```

Comment:
You can add a default case which will cover everything else
```suggestion
    switch (obscureLevel) {
      case "easy":
        return Destination.Obscurity.EASY;
      case "medium":
        return Destination.Obscurity.MEDIUM;
      case "hard":
        return Destination.Obscurity.HARD;
      default:
        return Destination.Obscurity.UNDEFINED; 
    }
```